clients/src/main/java/org/apache/kafka/common/security/oauthbearer/internals/OAuthBearerSaslClient.java

```diff
@@ -150,6 +154,20 @@ private void setState(State state) {
         this.state = state;
     }
 
+    private SaslExtensionsCallback parseCustomExtensions() throws SaslException {
+        SaslExtensionsCallback extensionsCallback = new SaslExtensionsCallback();
+        try {
+            callbackHandler().handle(new Callback[] {extensionsCallback});
+        } catch (UnsupportedCallbackException e) {
+            log.debug("Extensions callback is not supported by client callback handler {}, no extensions will be added",
+                    callbackHandler());
+        } catch (Throwable e) {
```

Comment:
Typically catch Exception rather than Throwable since Error generally should not be caught as it denots an event that the code can't really do anything about.